nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/test/java/org/apache/nifi/avro/TestAvroTypeUtil.java

```diff
@@ -587,4 +588,49 @@ public void testListAndMapConversion() {
             assertNotNull(((Record)inner).get("Message"));
         }
     }
+
+    @Test
+    public void testConvertToAvroObjectWhenFloatVSUnion_INT_FLOAT_ThenReturnFloat() {
```

Comment:
These two tests can be done more elegantly. Also need to add a version where INT is expected (and maybe test some other types).